tests/cupy_tests/random_tests/common_distributions.py

```diff
@@ -0,0 +1,225 @@
+import functools
+import unittest
+
+import numpy
+
+import cupy
+from cupy import testing
+from cupy.testing import _attr
+from cupy.testing import _condition
+
+
+def two_sample_Kolmogorov_Smirnov_test(observed1, observed2):
+    """Computes the Kolmogorov-Smirnov statistic on 2 samples
+
+    Unlike `scipy.stats.ks_2samp`, the returned p-value is not accurate
+    for large p.
+    """
+    assert observed1.dtype == observed2.dtype
+    n1, = observed1.shape
+    n2, = observed2.shape
+    assert n1 >= 100 and n2 >= 100
+    observed = numpy.concatenate([observed1, observed2])
+    indices = numpy.argsort(observed)
+    observed = observed[indices]  # sort
+    ds = numpy.cumsum(numpy.where(indices < n1, -n2, n1).astype(numpy.int64))
+    assert ds[-1] == 0
+    check = numpy.concatenate([observed[:-1] < observed[1:], [True]])
+    ds = ds[check]
+    d_plus = float(ds.max()) / (n1 * n2)
+    d_minus = -float(ds.min()) / (n1 * n2)
+    d = max(d_plus, d_minus)
+    # Approximate p = special.kolmogorov(d * numpy.sqrt(n1 * n2 / (n1 + n2)))
+    p = min(1.0, 2.0 * numpy.exp(-2.0 * d**2 * n1 * n2 / (n1 + n2)))
+    return d_plus, d_minus, p
+
+
+class BaseGeneratorTestCase(unittest.TestCase):
+
+    target_method = None
+
+    def get_random_state(self, xp, seed):
+        pass
+
+    def set_rs_seed(self, seed):
+        pass
+
+    def setUp(self):
+        self.__seed = testing.generate_seed()
+        self.rs = self.get_random_state(cupy, self.__seed)
```

Comment:
- Could you comment `self.rs` will be a (new or old) generator object.
- I'd rather not mixing `rs` and `random_state`.  My suggestion is to name them `rng`.

Reply:
done! abstracted parameter sets too

Reply:
How about renaming methods?
- `get_random_state` -> `get_rng`
- `set_rs_seed` -> `set_rng_seed` (or `reseed_rng`)

Reply:
done, sorry for overlooking it